coordinator/src/main/scala/filodb.coordinator/queryplanner/SingleClusterPlanner.scala

```diff
@@ -585,6 +588,15 @@ class SingleClusterPlanner(val dataset: Dataset,
       realScanStepMs, realScanEndMs, window, Some(execRangeFn), qContext,
       logicalPlanWithoutBucket.stepMultipleNotationUsed,
       paramsExec, logicalPlanWithoutBucket.offsetMs, rawSource)))
+
+    // Add the le filter transformer to select the required bucket
+    if (nameFilter.isDefined && nameFilter.head.endsWith("_bucket") && leFilter.isDefined) {
```

Comment:
``nameFilter.exists(_.endsWith("_bucket"))`` is more readable. Pattern match is even better

```
(nameFilter, leFilter) match {

   case Some(filter), Some(le) if filter.endsWith("_bucket") => // Your logic here, use variable le and filter directly, no need to use head/get on options

case _  =>  //NOP

}
```

Reply:
Sounds good @amolnayak311 !! Updating this ^